CT-1816 cannot assign SAR without message

misaka opened the pull request:
Fixes issues with being able to create SAR cases without entering case details, and with being able to view uploaded request attachments for a SAR case.

aliuk2012 commented on spec/site_prism/page_objects/pages/teams/show_page.rb:
```diff
@@ -0,0 +1,45 @@
+        element :team_lead_title, 'h2#team_lead_title'
+        element :team_lead_name, 'h3#team_lead_name'
+
+        section :table_heading, '.table-heading-row' do
```
get rid of the class and use thead tr

aliuk2012 commented on spec/site_prism/page_objects/pages/teams/show_page.rb:
```diff
@@ -0,0 +1,45 @@
+          element :num_subteams, 'th[area-label="Number of sub-teams column heading"]'
+        end
+
+        sections :team_list, '.team-row' do
```
tbody tr

aliuk2012 commented on spec/site_prism/page_objects/pages/users/index_page.rb:
```diff
@@ -0,0 +1,14 @@
+      class IndexPage < SitePrism::Page
+        set_url '/users/'
+
+        sections :users, '.user_row' do
```
tbody tr

aliuk2012 commented on spec/site_prism/page_objects/pages/users/index_page.rb:
```diff
@@ -0,0 +1,14 @@
+        set_url '/users/'
+
+        sections :users, '.user_row' do
+          element :full_name, 'td[aria-label="Full name"]'
```
aria-label="Name"

aliuk2012 commented on spec/site_prism/page_objects/pages/teams/show_page.rb:
```diff
@@ -0,0 +1,45 @@
+          element :actions, 'td[aria-label="Actions"]'
+        end
+
+        sections :information_officers, '#information_officers .user_row' do
```
.report tbody tr and change the view

aliuk2012 commented on app/views/teams/_business_unit_detail.html.slim:
```diff
@@ -0,0 +1,31 @@
+  | Information Officers
+.grid-row
+  .column-full.table-container.container
+    table#information_officers.table-font-xsmall
```
remove the id and add report class

aliuk2012 commented on app/views/teams/_business_unit_detail.html.slim:
```diff
@@ -30,3 +30,10 @@ h2.heading-medium
               | Edit
+.grid-row
+  .column-full
+    - if Pundit.policy(current_user, team).can_add_new_responder?
```
I would move this above line 33, no point outputting the extra markup if you not going to have any content in the grid-row

aliuk2012 commented on spec/views/teams/show_html_slim_spec.rb:
```diff
@@ -29,47 +30,92 @@ def login_as(user)
-      expect(bg.num_children.text).to eq '2'
+      expect(teams_show_page.heading.text)
+        .to have_text 'You are viewing Business group HMPPS'
+      bg = teams_show_page.directorates_list[0]
```
couldn't you grab the business group using row_for_business_group like the index page view spec does. https://github.com/ministryofjustice/correspondence_tool_staff/pull/309/files#diff-28b649368c6f972da4079daa71fc5285R34

aliuk2012 commented on spec/views/teams/show_html_slim_spec.rb:
```diff
@@ -29,47 +30,92 @@ def login_as(user)
-      expect(bg.name.text).to eq 'HR'
-      expect(bg.team_lead.text).to eq 'Martin Beecroft'
-      expect(bg.num_children.text).to eq '1'
+      bg = teams_show_page.directorates_list[1]
```
Same as above on line 38

misaka committed:
CT-1816 add test to create SAR case without a message

misaka committed:
CT-1816 fix validations for uploaded request files on SAR cases

misaka committed:
CT-1816 ensure both case details and attachments are visible

This fixes an issue with SAR cases where they weren't able to view request
attachments on the case details page.

misaka committed:
CT-1816 add attachments to tests that test the view
